### PR TITLE
Add all resource types to .travis.yml

### PR DESCRIPTION
**Pull request by ammokhov**
*Issue #, if available:*

*Description of changes:*

adding metric filter to a travis ci

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,14 +8,17 @@ env:
...
+    - script:
+        - cd "$TRAVIS_BUILD_DIR/aws-logs-loggroup"
+        - mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify
+      after_failure:
+        - cat "$TRAVIS_BUILD_DIR/aws-logs-loggroup/rpdk.log"
+    - script:
+        - cd "$TRAVIS_BUILD_DIR/aws-logs-metricfilter"
+        - mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify
+      after_failure:
+        - cat "$TRAVIS_BUILD_DIR/aws-macie-metricfilter/rpdk.log"
```
Comment (miparnisari) [inline .travis.yml:24]:
```suggestion
        - cat "$TRAVIS_BUILD_DIR/aws-logs-metricfilter/rpdk.log"
```

Comment (ammokhov) [reply .travis.yml:24]: oops

**Comment by PatMyron** [comment]
There are more [`AWS::Logs::` resource types](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/AWS_Logs.html). Can we get something more general for [all of these repositories](https://github.com/aws-cloudformation?q=aws-cloudformation-resource-providers-)?
```bash
for directory in $TRAVIS_BUILD_DIR/aws-*; do (cd "$directory" && mvn clean verify); done
```

**Comment by miparnisari** [comment]
@ammokhov can you implement Pat's suggestion? I really don't want to have to remember to change the travis config every time we add a new resource type :D

File: .travis.yml [hunk 2]
```diff
@@ -8,14 +8,11 @@ env:
...
-      mvn
-      -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
-      -B
-      clean verify
-after_failure:
-  - cat "$TRAVIS_BUILD_DIR/aws-logs-loggroup/rpdk.log"
+jobs:
+  include:
+    - script:
+        - for directory in $TRAVIS_BUILD_DIR/aws-*; do (cd "$directory" && mvn clean verify); done
```
Comment (PatMyron) [inline .travis.yml:16]: [left all the arguments out of my comment showing the general structure:](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/pull/42#issuecomment-651423690)
```suggestion
        - for directory in $TRAVIS_BUILD_DIR/aws-*; do (cd "$directory" && mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify); done
```

**Comment by miparnisari** [comment]
With the last commit, Travis is checking both resource types, but it fails with MetricFilter because it's using version 1.0.5. of the plugin and it mandates upgrading to 2.0.0. However, this update isn't trivial, 2 tests are breaking and I still don't know why.

**Comment by wbingli** [comment]
> With the last commit, Travis is checking both resource types, but it fails with MetricFilter because it's using version 1.0.5 of the Java plugin and it mandates upgrading to 2.0.0. However, this update isn't trivial, 2 tests are breaking and I still don't know why.

Could you have more details about the 2 tests fails?  Maybe good to have a PR to show the upgrade to 2.0.0 and the test fails?

**Comment by wbingli** [comment]
> With the last commit, Travis is checking both resource types, but it fails with MetricFilter because it's using version 1.0.5 of the Java plugin and it mandates upgrading to 2.0.0. However, this update isn't trivial, 2 tests are breaking and I still don't know why.

I took a look, there is the PR #46 why it's failing and the fix.

The `cloudformation-cli-java-plugin` 2.0.0 requires java dependency as 2.0.0.  You can pin the pip java plugin so the java pom.xml can stay in 1.0.4.  That means in the .travis.yml do this `pip3 install --user pre-commit cloudformation-cli-java-plugin==0.1.6`

**Comment by miparnisari** [comment]
Re-tried a build against master: https://travis-ci.com/github/aws-cloudformation/aws-cloudformation-resource-providers-logs/builds/174522848

We need to fix the .travis.yml file so that if any of the maven builds fail, the overall build fails too.

As it stands now, LogDestination is failing due to using 1.0.5. https://travis-ci.com/github/aws-cloudformation/aws-cloudformation-resource-providers-logs/builds/174522848#L434

File: .travis.yml [hunk 3]
```diff
@@ -8,14 +8,11 @@ env:
...
-      mvn
-      -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
-      -B
-      clean verify
-after_failure:
-  - cat "$TRAVIS_BUILD_DIR/aws-logs-loggroup/rpdk.log"
+jobs:
+  include:
+    - script:
+        - for directory in $TRAVIS_BUILD_DIR/aws-*; do (cd "$directory" && mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify); done
```
Comment (miparnisari) [inline .travis.yml:16]: Need to inspect the exit code of this and if any of the maven builds fail, we need to fail the entire build

**Comment by miparnisari** [comment]
The build is failing now, which is correct, but now i'm not sure how we're going to merge this PR if it's a required check

File: .travis.yml [hunk 4]
```diff
@@ -8,14 +8,12 @@ env:
...
   - pre-commit run --all-files --verbose
-  - cd "$TRAVIS_BUILD_DIR/aws-logs-loggroup"
-  # from Maven 3.6.1+, should use `--no-transfer-progress` instead of Slf4jMavenTransferListener
-  - >
-      mvn
-      -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn
-      -B
-      clean verify
-after_failure:
-  - cat "$TRAVIS_BUILD_DIR/aws-logs-loggroup/rpdk.log"
```
Comment (PatMyron) [inline .travis.yml:21]: do we still need these logs outputted after failures?

Comment (miparnisari) [reply .travis.yml:21]: 🤔 I removed that in the last commits

Comment (miparnisari) [reply .travis.yml:21]: [i tried keeping the command but it didn’t output anything due to the `travis_terminate 1` command](https://travis-ci.com/github/aws-cloudformation/aws-cloudformation-resource-providers-logs/jobs/358122689)

**Pull request merged**